types/search-grounding.ts

```diff
@@ -0,0 +1,165 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DynamicRetrievalMode } from "./enums";
+
+/**
+ * Retrieval tool that is powered by Google search.
+ * @public
+ */
+export declare interface GoogleSearchRetrievalTool {
+  /**
+   * Google search retrival tool config.
+   */
+  googleSearchRetrieval?: GoogleSearchRetrieval;
+}
+
+/**
+ * Retrieval tool that is powered by Google search.
+ */
+export declare interface GoogleSearchRetrieval {
+  /**
+   * Specifies the dynamic retrieval configuration for the given source.
+   */
+  dynamicRetrievalConfig?: DynamicRetrievalConfig;
+}
+
+/**
+ * Specifies the dynamic retrieval configuration for the given source.
+ */
+export declare interface DynamicRetrievalConfig {
+  /**
+   * The mode of the predictor to be used in dynamic retrieval.
+   */
+  mode?: DynamicRetrievalMode;
+  /**
+   * The threshold to be used in dynamic retrieval. If not set, a system default value is used.
+   */
+  dynamicThreshold?: number;
+}
+
+/**
+ * Metadata returned to client when grounding is enabled.
+ */
+export declare interface GroundingMetadata {
+  /**
+   * Google search entry for the following-up web searches.
+   */
+  searchEntryPoint?: SearchEntryPoint;
+  /**
+   * List of supporting references retrieved from specified grounding source.
+   */
+  groundingChuncks?: GroundingChunk[];
+  /**
+   * List of grounding support.
+   */
+  groundingSupport?: GroundingSupport[];
+  /**
+   * Metadata related to retrieval in the grounding flow.
+   */
+  retrievalMetadata?: RetrievalMetadata;
+  /**
+   * * Web search queries for the following-up web search.
+   */
+  webSearchQueries: string[];
+}
+
+/**
+ * Google search entry point.
+ */
+export declare interface SearchEntryPoint {
+  /**
+   * Web content snippet that can be embedded in a web page or an app webview.
+   */
+  renderedContent?: string;
+  /**
+   * Base64 encoded JSON representing array of <search term, search url> tuple.
+   */
+  sdkBlob?: string;
+}
+
+/**
+ * Grounding chunk.
+ */
+export declare interface GroundingChunk {
+  /**
+   *  Chunk from the web.
+   */
+  web?: GroundingChunkWeb;
+}
+
+/**
+ *  Chunk from the web.
+ */
+export declare interface GroundingChunkWeb {
+  /**
+   * URI reference of the chunk.
+   */
+  uri?: string;
+  /**
+   * Title of the chunk.
+   */
+  title?: string;
+}
+
+/**
+ *  Grounding support.
+ */
+export declare interface GroundingSupport {
+  /**
+   * URI reference of the chunk.
+   */
+  segment?: string;
+  /**
+   * A list of indices (into 'grounding_chunk') specifying the citations associated with the claim. For instance [1,3,4] means that grounding_chunk[1], grounding_chunk[3], grounding_chunk[4] are the retrieved content attributed to the claim.
```

Comment:
Nit: break long comment lines to fit the 80-char line width limit, e.g.
```
 /**
  * A list of indices (into 'grounding_chunk') specifying the citations
  * associated with the claim. For instance [1,3,4] means that
  * grounding_chunk[1], grounding_chunk[3], grounding_chunk[4] are the
  * retrieved content attributed to the claim.
  */
 ```
 
 Here and in other parts of this file.